### ISSUE TITLE
Capture programmatically a hardware trigger of a industrial camera

### ISSUE THREAD

**Issue by thanos91** [genicam/harvesters#117]
**Is your feature request related to a problem? Please describe.**
Is it possible to use the harvester module to listen to a hardware event like the pressing of the trigger button? If possible, i'd like to listen to such an event and when an image is acquired fetch it

**Comment by kazunarikudo** [comment]
@thanos91 Hi. Thank you for getting interested in Harvester. Of course, you can do that with Harvester. I’ll post a sample code later. Thank you for your patience. /Kazunari

**Comment by Frodo181** [comment]
Hi All,
I would also like to be able to control the camera in Python to set up the camera to acquire an image when it receives a hardware trigger. 

For me it is not clear from the section on Manipulating GenICam Feature Nodes where I can find the documentation on the properties listed with dir(ia.remote_device.node_map) and how to change them. Where would I be able to find the documentation for this?

**Comment by kazunarikudo** [comment]
@Frodo181 Hi. Thank you for trying out Harvester. In general, you can trigger the camera with the following code:
```
n = ia.remote_device.node_map  # Just to make a line shorter.
n.TriggerMode.value = "On"  # Turns trigger mode on.
n.TriggerSource.value = "Hardware"  # Use hardware trigger.
n.TriggerActivation.value = "RisingEdge"  # Could be omitted; triggers with the rising edge of the signal.
n.TriggerSource.value = "Line1"  # Use line #1 as the trigger source.
```
My intention for the section you referenced was to introduce major feature node types and how to read/write a value from/to the node. In the above code, they are all enumerations but you would have to work with other node types. Please feel free to ask me questions when needed. /Kazunari

**Comment by kazunarikudo** [comment]
GenICam defines a standard called Standard Feature Naming Convention, SFNC in short. I would recommend you to have a look at that when you can because it is fundamental. You can download the latest version 2.5 at [here](https://www.emva.org/wp-content/uploads/GenICam_SFNC_v2_5.pdf). The sections titled "Acquisition and Trigger features usage examples" and "Acquisition timing diagrams" may be a place where you should visit.

**Comment by Frodo181** [comment]
Thanks a lot, this should get me started!

**Comment by kazunarikudo** [comment]
You are welcome! It could be confusing in the beginning but you will notice each GenICam standard is consistent in the end. Please do not give up to logically follow each statement in the standards.

**Comment by VipulPandey-arch** [comment]
> You are welcome! It could be confusing in the beginning but you will notice each GenICam standard is consistent in the end. Please do not give up to logically follow each statement in the standards.

hello , i tried to connect basler line scan cam with pypylon,,i could detect the camera.but i am not able to take image and save it.
kindly see the issue

**Comment by prosemtui** [comment]
@kazunarikudo can you please post a sample code to catch images that are acquired by a hardware triger event, like you said in you answer to @thanos91 This would be highly apreciated as a base to start from.

**Comment by aspini** [comment]
@kazunarikudo 
I need to define manually the trigger. 
I saw your example:
'''
n = ia.remote_device.node_map  # Just to make a line shorter.
n.TriggerMode.value = "On"  # Turns trigger mode on.
n.TriggerSource.value = "Hardware"  # Use hardware trigger.
n.TriggerActivation.value = "RisingEdge"  # Could be omitted; triggers with the rising edge of the signal.
n.TriggerSource.value = "Line1"  # Use line #1 as the trigger source
'''

I need to define n.TriggerMode.value, n.TriggerSource.value and n.TriggerSource.value so that the camera captures at a distance of an a pre-defined time interval.
For example, once I defined the time interval = 0.5 seconds I need the camera to acquire exactly every 0.5 seconds. 
My doubts are: 
1)	can I generate this trigger solution? I didn’t find how to customize trigger variables inside core.py and geneapi.py.
2)	If the answer of the previous point is positive and  the camera is set to 4 fps, i.e. the interval between each frame is 0.25 seconds and I set the camera to acquire after 0.55 seconds (my dt is always 0.5 sec) from image_start_acquisition(), i.e. I’m between second and third frame, do I have to wait 0.2 sec to get my acquisition? 
Or, once I defined the trigger mode, which in this case it’s every 0.5 sec, am I sure that I will acquire exactly in that instant?
(P.S.  Can I modify the frame rate from node_map variables?
